[wasm] Fix order of include paths, to have the obj dir first

radical opened the pull request:
this regressed recently, and we started using `pinvoke-table.h`, and
`icall-table.h` from the runtime pack instead of from the obj dir.

To fix:
- `pinvoke-table.h` should not be included in the runtime pack anyway, so remove that
- Fix the order of the include paths

radical committed:
[wasm] Fix order of include paths, to have the obj dir first

this regressed recently, and we started using `pinvoke-table.h`, and
`icall-table.h` from the runtime pack instead of from the obj dir.

radical commented:
I'll add a test for this once https://github.com/dotnet/runtime/pull/49398 is merged.

vargaz commented:
Maybe we should rename those files, and include them inside the ifdef to prevent this from happening.

lewing commented:
I'm pretty sure this pinvoke-table.h _is_ the renamed file, but for some reason it is also in the runtime pack (along side pinvoke.h)

radical commented:
So, should we:

1. remove `pinvoke-table.h` from the runtime pack,
2. and always use the generated `pinvoke-table.h` ?

lewing commented:
I looked closer and it appears I was wrong.  If we want to rename that is fine, double check what is happening with the icall tables as well.

radical committed:
[wasm] Don't include pinvoke-table.h in the runtime pack.

`pinvoke-table.h` is generated when building the native files for wasm.
They are not required in the runtime pack, as any wasm app build needing
to do native relinking would be generating one of it's own.

radical committed:
Fix the property name

radical commented:
> Maybe we should rename those files, and include them inside an ifdef to prevent this from happening.

IIUC, `pinvoke-table.h` is generated when building the native files for wasm.
They are not required in the runtime pack, as any wasm app build needing
to do native relinking would be generating one of its own.

So, I removed it from the runtime pack, and the rename shouldn't be needed.
